include/fs_utils.h

```diff
@@ -32,6 +32,37 @@
 
 #include "std_filesystem.h"
 
+#if defined(WIN32)
+#include <windows.h>
+using NativeFileHandle = HANDLE;
+// Cannot be constexpr due to Win32 macro
+#define InvalidNativeFileHandle INVALID_HANDLE_VALUE
+#else
+using NativeFileHandle = int;
+constexpr NativeFileHandle InvalidNativeFileHandle = -1;
+#endif
+
+constexpr int32_t NativeFileSeekFailed = -1;
+
+struct NativeFileReturn
+{
+	uint32_t bytes = 0;
+	bool error = false;
+};
+
+enum class NativeFileSeek
+{
+	Set,
+	Current,
+	End
+};
+
+struct DosDateTime
+{
+	uint16_t date = 0;
+	uint16_t time = 0;
+};
```

Comment:
Just run the formatter on these. Then you'll get stuff like this:

```cpp
enum class NativeFileSeek { Set, Current, End };
```

Reply:
formatted